### ISSUE TITLE
Allow server to continue running during rebuilds

### ISSUE THREAD

**Issue by dobesv** [meteor/meteor-feature-requests#303]
It would be great if the server would continue running while rebuilding.

Currently if I make a change, the server stops, so I can't do much testing or messing around in the app until it finishes rebuilding and restarting the server.

**Comment by dr-dimitru** [comment]
Hello @dobesv are talking about production here?

**Comment by dobesv** [comment]
No, development.

**Comment by dr-dimitru** [comment]
@dobesv how long does it take to rebuild an app?

**Comment by dr-dimitru** [comment]
@dobesv this is a lot!

 1. What stack are you on? (Blaze, React, Vue, other)
 2. Could you post your `package.json` file?

There is a way to speed up building time by installing some NPM packages.
Make sure you have globally and locally installed `node-gyp` and `node-pre-gyp`.

**Comment by dobesv** [comment]
Hi, there's already some discussion about it in a meteor issue, I think
it's a bit separate from this feature request.  Sounds like something to do
with there being a lot of files in node_modules.

https://github.com/meteor/meteor/issues/9969

How long does it take for meteor to rebuild on your project(s) ?

**Comment by dr-dimitru** [comment]
@dobesv biggest projects is ~10 sec and I'm on old 2012 MBP with SSD, but even SSD is old and not that fast as new macs at all.
Small average size almost instant for Client and 1-3 sec for Server rebuild.
Including projects where we use TS, Sass, Jade, CoffeeScript - build time usually a bit longer but, still not that long.

**Comment by dobesv** [comment]
Yeah, I don't think it's the number of technologies you use.  We must be
importing some packages that have a lot of dependencies.  Like Apollo,
DraftJS, etc..  I don't know exactly which one is the issue.

**Comment by dr-dimitru** [comment]
@dobesv sorry for late reply, yes, many ways:
 - Via `npm install` module would be left as it is, so in case if ES6/7 used in module, it wouldn't b processed with Babel
 - Via `npm link`
 - Fork/Download lib to Meteor's project directory, more from [Meteor's releases](https://github.com/meteor/meteor/blob/devel/History.md):

Although Meteor does not recompile packages installed in `node_modules`
  by default, compilation of specific npm packages (for example, to
  support older browsers that the package author neglected) can now be
  enabled in one of two ways:

  * Clone the package repository into your application's `imports`
    directory, make any modifications necessary, then use `npm install` to
    link `the-package` into `node_modules`:
    ```sh
    meteor npm install imports/the-package
    ```
    Meteor will compile the contents of the package exposed via
    `imports/the-package`, and this compiled code will be used when you
    import `the-package` in any of the usual ways:
    ```js
    import stuff from "the-package"
    require("the-package") === require("/imports/the-package")
    import("the-package").then(...)
    ```
    This reuse of compiled code is the critical new feature that was added
    in Meteor 1.7.

  * Install the package normally with `meteor npm install the-package`,
    then create a symbolic link *to* the installed package elsewhere in
    your application, outside of `node_modules`:
    ```sh
    meteor npm install the-package
    cd imports
    ln -s ../node_modules/the-package .
    ```
    Again, Meteor will compile the contents of the package because they
    are exposed outside of `node_modules`, and the compiled code will be
    used whenever `the-package` is imported from `node_modules`.

    > Note: this technique also works if you create symbolic links to
      individual files, rather than linking the entire package directory.

  In both cases, Meteor will compile the exposed code as if it was part of
  your application, using whatever compiler plugins you have installed.
  You can influence this compilation using `.babelrc` files or any other
  techniques you would normally use to configure compilation of
  application code. [PR #9771](https://github.com/meteor/meteor/pull/9771)
  [Feature #6](https://github.com/meteor/meteor-feature-requests/issues/6)

  > ~Note: since compilation of npm packages can now be enabled using the
    techniques described above, Meteor will no longer automatically scan
    `node_modules` directories for modules that can be compiled by
    compiler plugins. If you have been using that functionality to import
    compiled-to-JS modules from `node_modules`, you should start using the
    symlinking strategy instead.~ **Follow-up note: this optimization was
    reverted in Meteor 1.7.0.1 (see [above](#v1701-2018-05-29)).**
